File: Erik_Vincent/Assignment1/Assignment1/Model/QLS/AST/DefaultStyle.cs
using System.Collections.Generic;
using Type = Assignment1.Model.QL.AST.Type;
using Assignment1.Model.QLS.AST.Style;

namespace Assignment1.Model.QLS.AST
{
    public class DefaultStyle : Statement
    {
        public Type Type { get; }
        public IReadOnlyCollection<IStyle> Styles => _styles.AsReadOnly();

        private readonly List<IStyle> _styles;

        public DefaultStyle(Type type, List<IStyle> styles)
        {
            Type = type;
            _styles = styles;
        }

        public override void Accept(IQLSASTVisitor visitor) => visitor.Visit(this);
    }
}

File: Remi_Hasan/src/main/java/qls/model/statement/DefaultStyle.java
package qls.model.statement;

import ql.model.expression.ReturnType;
import qls.model.style.StyleAttribute;
import qls.model.widget.Widget;
import qls.visitor.IQLSVisitor;

import java.util.List;

public class DefaultStyle extends Statement {

    private final ReturnType type;
    private final List<StyleAttribute> styleAttributes;
    private final Widget widget;

    public DefaultStyle(ReturnType type, List<StyleAttribute> styleAttributes, Widget widget) {
        this.type = type;
        this.styleAttributes = styleAttributes;
        this.widget = widget;
    }

    public ReturnType getType() {
        return type;
    }

    public List<StyleAttribute> getStyleAttributes() {
        return styleAttributes;
    }

    public Widget getWidget() {
        return widget;
    }

    @Override
    public <T> T accept(IQLSVisitor<T> visitor) {
        return visitor.visit(this);
    }
}
